examples/cross_decomposition/plot_pcr_vs_pls.py

```diff
@@ -0,0 +1,151 @@
+toy dataset. Our goal is to illustrate how PLS can outperform PCR when the
+target is strongly correlated with some directions in the data that have a
+low variance.
+
+PCR is a regressor composed of two steps: first,
+:class:`~sklearn.decomposition.PCA` is applied to the training data, possibly
+performing dimensionality reduction; then, a regressor (e.g. a linear
+regressor) is trained on the transformed samples. In
+:class:`~sklearn.decomposition.PCA`, the transformation is purely
+unsupervised, meaning that no information about the targets is used. As a
+result, PCR may perform poorly in some datasets where the target is strongly
+correlated with *directions* that have low variance. Indeed, the
+dimensionality reduction of PCA projects the data into a lower dimensional
+space where the variance of the projected data is greedily maximized along
+each axis. Despite them having the most predictive power on the target, the
+directions with a lower variance will be dropped, and the final regressor
+will not be able to levarage them.
+
+PLS is both a transformer and a regressor, and it is quite similar to PCR: it
+also applies a dimensionality reduction to the samples before applying a
+linear regressor to the transformed data. The main difference with PCR is
+that the PLS transformation is supervised. Therefore, as we will see in this
+example, it does not suffer from the issue we just mentioned.
+"""
+
+print(__doc__)
+
+##############################################################################
+# The data
+# --------
+#
+# We start by creating a simple dataset with two features. Before we even dive
+# into PCR and PLS, we fit a PCA estimator to display the two principal
+# components of this dataset, i.e. the two directions that explain the most
+# variance in the data.
+import numpy as np
+import matplotlib.pyplot as plt
+from sklearn.decomposition import PCA
+
+rng = np.random.RandomState(0)
+n_samples = 500
+cov = [[3, 3],
+       [3, 4]]
+X = rng.multivariate_normal(mean=[0, 0], cov=cov, size=n_samples)
+pca = PCA(n_components=2).fit(X)
+
+
+plt.scatter(X[:, 0], X[:, 1], alpha=.3, label='samples')
+for i, (comp, var) in enumerate(zip(pca.components_, pca.explained_variance_)):
+    comp = comp * var  # scale component by its variance explanation power
+    plt.plot([0, comp[0]], [0, comp[1]], label=f"Component {i}", linewidth=5)
+plt.gca().set(aspect='equal',
+              title="2-dimensional dataset with principal components")
+plt.legend()
+plt.show()
+
+##############################################################################
+# For the purpose of this example, we now define the target `y` such that it is
+# strongly correlated with a direction that has a small variance. To this end,
+# we will project `X` onto the second component, and add some noise to it.
+# Our goal is to illustrate that PCR will perform poorly on this dataset,
+# comparatively to PLS.
```

Comment:
A bit redundant, this is specified in the introduction and explained in the dedicated section.